File: Spring Boot/monolith/src/main/java/pl/edu/pg/benchmarking/monolith/place/Place.java
package pl.edu.pg.benchmarking.monolith.place;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import pl.edu.pg.benchmarking.monolith.distance.Point;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

@Getter
@AllArgsConstructor(access = AccessLevel.PUBLIC)
public class Place {

    private Integer id;

    private String city;

    private String coordinates;

    public Point getPoint() {
        Pattern pattern = Pattern.compile("(?<latitude>\\d+\\.?\\d*),\\s{1}(?<longitude>\\d+\\.?\\d*)");
        Matcher matcher = pattern.matcher(this.coordinates);
        matcher.find();
        Double latitude = Double.valueOf(matcher.group("latitude"));
        Double longitude = Double.valueOf(matcher.group("longitude"));
        return new Point(latitude, longitude);
    }

}

File: Flask/monolith/website/place/place.py
import re
from typing import Final
from static.utils import attributes
from website.distance.point import Point


COORDINATE_REGEX: Final[str] = "([0-9]+.[0-9]+), ([0-9]+.[0-9]+)"

class Place():
    def __init__(self, place_id: int, city: str, coordinates: str) -> None:
        self.id = place_id
        self.city = city
        self.coordinates = coordinates

    def as_dict(self) -> dict:
        return attributes(self)

    def get_point(self) -> Point:
        result = re.search(COORDINATE_REGEX, self.coordinates)
        latitude = float(result.group(1))
        longitude = float(result.group(2))
        return Point(latitude, longitude)
